Samples/auth/Outlook-Add-in-SSO-NAA-Identity/README.md

```diff
@@ -0,0 +1,276 @@
+
+All APIs should publish a minimum of one [App role for applications](https://docs.microsoft.com/azure/active-directory/develop/howto-add-app-roles-in-azure-ad-apps#assign-app-roles-to-applications), also called [Application Permission](https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#permission-types), for the client apps to obtain an access token as themselves, i.e. when they are not signing-in a user. Application permissions are the type of permissions that APIs should publish when they want to enable client applications to successfully authenticate as themselves and not need to sign-in users. To publish an application permission, follow these steps:
+
+1. Still on the same app registration, select the **App roles** blade to the left.
+1. Select **Create app role**:
+
+    ![app roles](./images/create-app-roles.png)
+
+    1. For **Display name**, enter a suitable name for your application permission, for example `Todolist.Read.All`.
+    1. For **Allowed member types**, choose **Application** to ensure other applications can be granted this permission.
+    1. For **Value**, enter `Todolist.Read.All`.
+    1. For **Description**, enter `Allows the app to read the signed-in user's files.`
+    1. To save your changes, select **Apply**.
+
+Repeat the previous steps to add another app permission named `Todolist.ReadWrite.All`.
+Your app roles should appear as shown in the following screenshot.
+
+![app roles complete](./images/app-roles-complete.png)
+
+### Configure the service app (Contoso-Web-API-Server) to use your app registration
+
+Open the sample project in Visual Studio Code to configure the code. In the following steps, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `API/server-helpers/authConfig.js` file.
+1. Find the key `Enter_API_Application_Id_Here` and replace the existing value with the application ID (clientId) of Contoso-Web-API-Server app copied from the Microsoft Entra admin center.
+1. Save the file.
+
+### Register the client app (Contoso-Outlook-Add-in)
+
+1. Go to the [Azure portal - App registrations](https://go.microsoft.com/fwlink/?linkid=2083908) page to register your app.
+1. Sign in with the **_admin_** credentials to your Microsoft 365 tenancy. For example, **MyName@contoso.onmicrosoft.com**.
+1. Select **New registration**. On the **Register an application** page, set the values as follows.
+
+   - For **Name**, enter `Contoso-Outlook-Add-in`.
+   - For **Supported account types**, select **Accounts in this organizational directory only**.
+   - Select **Register**.
+
+1. In the **Overview** blade, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
+1. In the app's registration screen, select the **Authentication** blade on the left pane.
+1. If you don't have a platform added, select **Add a platform** and choose the **Single-page application** option.
```

Comment:
Why would you have a platform added at this point in the sample?

Reply:
fixed

Reply:
fixed